cli/client.go

```diff
@@ -1197,3 +1209,45 @@ func (c *viamClient) startRobotPartShell(
 	outputLoop()
 	return nil
 }
+
+func parseLogEntryFields(fieldsRaw []*structpb.Struct) ([]zap.Field, error) {
+	parsedFields := make([]zap.Field, 0, len(fieldsRaw))
+	bytes, err := json.Marshal(fieldsRaw)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(bytes, &parsedFields); err != nil {
+		return nil, err
+	}
+	return parsedFields, nil
+}
+
+func logEntryFieldsToString(fields []zap.Field) string {
+	fieldsString := ""
+	for _, field := range fields {
+		fieldValue := ""
+		// TODO use zap encoder to encode
+		// it is a bit hard to do that though because sending the data over the wire
+		// loses type info which causes the zap encoders to panic.
+		// This code is modeled after zapcore.Field.AddTo
+		//nolint:exhaustive
```

Comment:
Needed with the `default` case?

Reply:
I don't think we have `defaultSignifiesExhaustive` set on our linter (defaults to false) https://golangci-lint.run/usage/linters/#exhaustive

Reply:
That'd be it! I like leaving it as false; thanks for looking that up.